_test/test_transformation/test_ortho_proj_construction.m

```diff
@@ -0,0 +1,247 @@
+            assertExceptionThrown(...
+                @()ortho_proj([1,0,0],[1,0,0],'alatt',[2,3,4],'angdeg',[80,70,85]),...
+                'HORACE:ortho_proj:invalid_argument');
+        end
+        
+        function test_set_wrong_u(~)
+            proj = ortho_proj([1,0,0],[0,1,0],'alatt',[2,3,4],'angdeg',[80,70,85]);
+            proj.u = [0,1,0];
+            assertTrue(ischar(proj.u));
+            assertExceptionThrown(@()isvalid(proj),'HORACE:ortho_proj:runtime_error');
+        end
+        function test_serialization(~)
+            proj = ortho_proj([1,0,0],[0,1,0],'alatt',[2,3,4],'angdeg',[80,70,85]);
+            
+            ser = proj.serialize();
+            rec = serializable.deserialize(ser);
+            
+            assertEqual(proj,rec);
+        end
+        
+        %------------------------------------------------------------------
+        function test_default_constructor(~)
+            proj = ortho_proj();
+            assertElementsAlmostEqual(proj.u,[1,0,0])
+            assertElementsAlmostEqual(proj.v,[0,1,0])
+            assertTrue(isempty(proj.w))
+            assertElementsAlmostEqual(proj.offset,[0,0,0,0])
+            assertEqual(proj.type,'aaa')
+            full_box = expand_box([0,0,0,0],[1,1,1,1]);
+            pixi = proj.transform_pix_to_img(full_box );
+            %assertElementsAlmostEqual(full_box,pixi);
+            pixp = proj.transform_img_to_pix(pixi);
+            assertElementsAlmostEqual(full_box,pixp);
+        end
+        %
+        function test_invalid_constructor_throw(~)
+            f = @()ortho_proj([0,1,0]);
+            assertExceptionThrown(f,'HORACE:ortho_proj:invalid_argument');
+        end
+        %
+        function test_uv_set_in_constructor(~)
+            proj = ortho_proj([0,1,0],[1,0,0]);
+            assertElementsAlmostEqual(proj.u,[0,1,0])
+            assertElementsAlmostEqual(proj.v,[1,0,0])
+            assertTrue(isempty(proj.w))
+            assertEqual(proj.type,'ppr')            
+        end
+        %
+        function test_uvw_set_in_constructor(~)
+            proj = ortho_proj([1,0,0],[0,1,1],[0,-1,1]);
+            assertElementsAlmostEqual(proj.u,[1,0,0])
+            assertElementsAlmostEqual(proj.v,[0,1,1])
+            assertElementsAlmostEqual(proj.w,[0,-1,1])
+        end
+        function test_get_set_from_data_matrix_ppp(~)
+            proj1 = ortho_projTester([1,0,0],[0,1,0],[0,0,1],...
+                'alatt',[2,4,3],'angdeg',[90,90,90],...
+                'label',{'a','b','c','d'},'type','ppp');
+            
+            [~, u_to_rlu, ulen] = proj1.projaxes_to_rlu_public([1,1,1]);
+            
+            pror = ortho_projTester('alatt',[2,4,3],'angdeg',[90,90,90],...
+                'label',{'a','b','c','d'});
+            pror = pror.set_from_data_mat(u_to_rlu,ulen);            
+            % TODO: This is something wrong as inputs are not recovered. 
+            % Is this correct? Should (can I fix this)
```

Comment:
this is all about uncertain projection type `ortho_proj.type`.  I am not sure I am recovering projection from existing data in any case. I am hoping to avoid this issue storing the projection in `sqw`, but until that I can not be sure I am doing all properly. 
This will request more attention if we decided to make release in old format (I hope not)